crypto/batchverifier.go

```diff
@@ -111,6 +112,18 @@ func (b *BatchVerifier) getNumberOfEnqueuedSignatures() int {
 // Verify verifies that all the signatures are valid. in that case nil is returned
 // if the batch is zero an appropriate error is return.
 func (b *BatchVerifier) Verify() error {
+	return b.VerifyWithFeedback(nil)
+}
+
+// VerifyWithFeedback verifies that all the signatures are valid.
+// failed slice should have len equal to getNumberOfEnqueuedSignatures
+// if all sigs are valid, nil will be returned
+// if some txns are invalid, true will be set at the appropriate index in failed
+func (b *BatchVerifier) VerifyWithFeedback(failed []bool) error {
```

Comment:
I agree that is a good idea to have two `verify` functions However, I think it would be much simpler to just return `failed` array

Reply:
I can't agree more.